Fixed personal id date validation

krynecouk opened the pull request:
Personal id date validation was mistakenly accepting invalid dates and ignoring leap years. This commit fix this bug.

Tomucha commented:
Hele, Dárek! Proč si to nenapsal rovnou pořádně ty mezuláne, co?

Tak lepší později než nikdy, že, díky že na mě tak myslíš.

Když teda to vylepšujem, tak _isValidDate je nějaká podivná teda - proč to lepit do stringu a pak parsovat? DateTime má přímo konstruktor co bere y,m,d, ... Tak to udělej přes to, ne? Akorát si nejsem jistej, jestli náhodou když mu dáš třeba M=13, tak z toho neudělá M=1 a Y o jedna zvedne ... možná, nevim, bude to chtít test case.

krynecouk committed:
Fixed personal id date validation

Personal id date validation was mistakenly accepting invalid dates and ignoring leap years. This commit fix this bug.

krynecouk commented:
~~On to prave zvedne, no. Kvuli tomu tam lepim ten string abych ho mohl porovnat s tou datetime hodnotou. Asi bych mohl kontrolovat jen ten mesic ale tohle je sichrovejsi...~~

Predelal jsem to jen na tu kontrolu mesice. Testy projdou, tak by to melo byt ok.

krynecouk committed:
Refactored _isValidDate

_isValidDate now checks only month and not whole date.

Tomucha commented:
https://pub.dev/packages/czech
